browser/mapping_test.go

```diff
@@ -193,6 +193,91 @@ func TestMappings(t *testing.T) {
 	}
 }
 
+type mockBrowserType struct {
```

Comment:
I think this has value, as it's testing a "hidden" behaviour, which can only be asserted by looking at the internals of what the `launch` method is doing based on some input params, otherwise the result of the `launch` is the same, a `browser`, so difficult to assert that nay changed.

We could improve the testability by using a behavioural pattern that is a bit more explicit in what it does based on the input, but that also complicates the implementation.

Reply:
I took out a piece of code temporarily:

https://github.com/grafana/xk6-browser/blob/b2a884cf80c2fb09ac0f33859632bbdb81325491/browser/mapping.go#L722-L725

And then executed [the integration test](https://github.com/grafana/xk6-browser/blob/b2a884cf80c2fb09ac0f33859632bbdb81325491/tests/browser_type_test.go) and it failed:

```
=== RUN   TestBrowserTypeLaunchToConnect
    browser_type_test.go:66: 
                Error Trace:    /Users/inanc/grafana/k6browser/main/tests/browser_type_test.go:66
                Error:          Should be true
                Test:           TestBrowserTypeLaunchToConnect
```

I thought the error message could be improved. Maybe something like:

```
When "K6_BROWSER_WS_URL" environment variable is provided, the launch method in the mapping layer should call BrowserType.Connect.
```

Anyway. I've checked the test code:

https://github.com/grafana/xk6-browser/blob/b2a884cf80c2fb09ac0f33859632bbdb81325491/tests/browser_type_test.go#L57-L66

Where I see it calls `chromium.Launch` and it does not connect. The problem is obvious. I'll first check the mapping layer since the test code has ``jsMod.Chromium.Launch`. So I looked at the mapping layer, found [the launch mapping](https://github.com/grafana/xk6-browser/blob/b2a884cf80c2fb09ac0f33859632bbdb81325491/browser/mapping.go#LL718), and fixed the problem.

I quickly located the problem with the integration test without leaning on the unit test. The unit test has a value, but I don't think it's necessary in this situation. It just told me `Connect` is not called. I already know that from the integration test.

The actual problem arose from the fact that the `launch -> connect` logic was put in the mapping layer. This made testing difficult, and the unit test became brittle: The test can fail with even a small change in the tested code. Instead, placing this logic in the inner layers could make testing easier.

Then again, I'm happy to go with what you guys decide.

Reply:
If you guys are happy with the integration test, I think we can remove this unit test at the mapping level.

Reply:
I'm happy with the integration test 😊